open-craft/cohort-manager
# SE-1406  select courses from ui

This adds a set of UI/UX improvements with the main goal to remove the dependency of a crafted url from the instructor dashboard of a course.

- add dropdown menus to select courses (no more manual entering course ids)
- make UI layout neater and clearer
- disable import button if not ready to import
- add loading spinner
- update variable naming conventions
- dedupe code
- remove experimental instructor dashboard plugin

**Screenshots**:

![1591579742](https://user-images.githubusercontent.com/9714796/83985235-883b2000-a977-11ea-9743-ab7a9e43bbd4.png)

![1591579699](https://user-images.githubusercontent.com/9714796/83985237-8a9d7a00-a977-11ea-8b44-50fd07c1e044.png)

![1591579672](https://user-images.githubusercontent.com/9714796/83985239-8bcea700-a977-11ea-8004-5bdac4feb127.png)

**Test instructions**:

- set up cohorts on at least two courses
- visit http://localhost:18000/cohort-manager/
- play with the UI and try to break it - import cohorts between courses, add/remove cohorts manually, try again, etc.
- check that the UI is usable with no gotchas that could cause importing cohorts into the wrong course, etc.

**Author notes/concerns**:

Due to time/budget constraints, the following features that would be nice have not been implemented:
- verify that courses have cohorts enabled before allowing import
- full error handling when xhr requests fail


**Reviewer**:
